features/facebookImport/src/components/buttons/routeButton.jsx

```diff
@@ -9,21 +9,16 @@ const RouteButton = ({
     children,
     onClick = () => {},
 }) => {
-    const { handleBack, changeNavigationState, navigationState } =
-        useContext(ImporterContext);
     const history = useHistory();
 
-    let changedNavigationState = navigationState;
+    const newHistoryState = { ...INITIAL_HISTORY_STATE, ...stateChange };
 
     const onClickButton = () => {
+        if (!route) return;
+
         onClick();
-        if (stateChange)
-            changedNavigationState = { ...navigationState, ...stateChange };
-        if (route) {
-            if (route == "back") handleBack();
-            else history.push(route, changedNavigationState);
-        }
-        changeNavigationState(changedNavigationState);
+        if (route == "back" && history.length > 1) history.goBack();
+        else history.push(route, newHistoryState);
```

Comment:
OK, so navigationState is not *really* removed. It's removed as a global variable. You're using history to keep the navigation state, reduced now to a simple scrolling state. If you need any other information (say, clicked state or whatever) you can still keep it there, and it will be local-ish to the route. That's sensible :clap: You eliminate possible inconsistencies.